src/main/java/dev/morling/onebrc/CalculateAverage_karthikeyan97.java

```diff
@@ -0,0 +1,348 @@
+                 * hascode = 1;
+                 * isNumber = false;
+                 * MeasurementAggregator agg = resultmap.get(stationName);
+                 * if (agg == null) {
+                 * agg = new MeasurementAggregator();
+                 * agg.min = num * sign;
+                 * agg.max = num * sign;
+                 * agg.sum = (long) (num * sign);
+                 * agg.count = 1;
+                 * resultmap.put(stationName, agg);
+                 * }
+                 * else {
+                 * agg.min = Math.min(agg.min, num * sign);
+                 * agg.max = Math.max(agg.max, num * sign);
+                 * agg.sum += (long) (num * sign);
+                 * agg.count++;
+                 * }
+                 * num = 1;
+                 * bbstart = buffer.position();
+                 * }
+                 * else {
+                 * hascode = 31 * hascode + bi;
+                 * if (isNumber) {
+                 * switch (bi) {
+                 * case 0x2E:
+                 * break;
+                 * case 0x2D:
+                 * num = num * -1;
+                 * break;
+                 * default:
+                 * num = num * 10 + (bi - 0x30);
+                 * }
+                 * }
+                 * }
+                 * }
+                 * }
+                 * }
+                 */
+                return resultmap;
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+            return null;
+        }).flatMap(e -> e.entrySet().stream()).collect(groupingBy(e -> e.getKey(), collector)))) {
+            @Override
+            public Object put(Object key, Object value) {
+                return super.put(((modifiedbytearray) key).getStationName(), value);
+            }
+        });
+
+        /*
+         * .map(a -> {
+         * return a.stream().parallel().collect(groupingBy(m -> m.station(), collector));
+         * }).flatMap(m -> m.entrySet()
+         * .stream()
+         */
+        // Get the FileChannel from the FileInputStream
+
+        //System.out.println("time taken:" + (System.nanoTime() - start) / 1000000);
+        // System.out.println(measurements);
+    }
+
+}
+
+class modifiedbytearray {
+    private int start;
+    private int end;
+    private ByteBuffer bb;
+    public int hashcode;
+
+    modifiedbytearray(int start, int end, int hashcode, ByteBuffer bb) {
+        this.start = start;
+        this.end = end;
+        this.bb = bb;
+        this.hashcode = hashcode;
+    }
+
+    public String getStationName() {
+        int length = end - start;
+        byte[] arr = new byte[length];
+        bb.get(start, arr, 0, length);
+        return new String(arr, StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public String toString() {
+        return getStationName();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        return ((modifiedbytearray) obj).hashcode == this.hashcode;
```

Comment:
Hum, how do you deal with collisions of the hash?

Reply:
fixed this